qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -193,43 +306,146 @@ def _default_options(self):
+                    point_q = q_center + qubit_iq_rand_sample[1]
+
+                    # Adding phase if not 0.0
+                    if not np.allclose(phase, 0.0):
+                        complex_iq = (point_i + 1.0j * point_q) * np.exp(1.0j * phase)
+                        point_i, point_q = np.real(complex_iq), np.imag(complex_iq)
+
+                    shot_memory.append([point_i, point_q])
+                # We proceed to the next occurrence - meaning it's a new shot.
+                memory.append(shot_memory)
+                shot_num += 1
 
         return memory
 
-    @abstractmethod
-    def _compute_probability(self, circuit: QuantumCircuit) -> float:
-        """Compute the probability used in the binomial distribution creating the IQ shot.
-
-        An abstract method that subclasses will implement to create a probability of
-        being in the excited state based on the received quantum circuit.
-
+    def _generate_data(
+        self, prob_dict: Dict[str, float], num_qubits: int, circuit: QuantumCircuit
+    ) -> Dict[str, Any]:
+        """
+        Generate data for the circuit.
         Args:
-            circuit: The circuit from which to compute the probability.
+            prob_dict(dict): A dictionary whose keys are strings representing the output vectors and
+            their values are the probability to get the output in this circuit.
+            num_qubits(int): The number of qubits.
+            circuit(QuantumCircuit): The circuit that needs to be simulated.
 
         Returns:
-             The probability that the binaomial distribution will use to generate an IQ shot.
+            A dictionary that's filled with the simulated data. The output format is different between
+            measurement level 1 and measurement level 2.
         """
+        self._verify_parameters(num_qubits, prob_dict)
+        prob_arr = self._probability_dict_to_probability_array(prob_dict, num_qubits)
+        shots = self.options.get("shots")
+        meas_level = self.options.get("meas_level")
+        meas_return = self.options.get("meas_return")
+        run_result = {}
+
+        if meas_level == MeasLevel.CLASSIFIED:
+            counts = {}
+            results = self._rng.multinomial(shots, prob_arr, size=1)[0]
+            for result, num_occurrences in enumerate(results):
+                result_in_str = str(format(result, "b").zfill(num_qubits))
+                counts[result_in_str] = num_occurrences
+            run_result["counts"] = counts
+        else:
+            phase = self._iq_phase(circuit)
+            memory = self._draw_iq_shots(prob_arr, shots, num_qubits, phase)
+            if meas_return == "avg":
+                memory = np.average(np.array(memory), axis=0).tolist()  # could have a bug here
+
+            run_result["memory"] = memory
+        return run_result
 
     # pylint: disable=unused-argument
     def _iq_phase(self, circuit: QuantumCircuit) -> float:
```

Comment:
Do we want to be symmetrical and convert it to a callable parameter of `__init__`, like we did for the function that computes probabilities?

Reply:
The `_iq_phase` function have moved into `mock_iq_helpers.py`